src/no-side-effects.ts

```diff
@@ -0,0 +1,116 @@
+// no-side-effects.ts
+
+/*
+ * Copyright (c) 2022-2024 Check Digit, LLC
+ *
+ * This code is licensed under the MIT license (see LICENSE.txt for details).
+ */
+
+import type { Node } from 'estree';
+import type { Rule } from 'eslint';
+
+const EXCLUDED_IDENTIFIERS = ['assert', 'debug', 'log'];
+
+function isTestFile(fileName: string): boolean {
```

Comment:
there is no need for this in the rule itself.  whether or not a rule runs against test files is part of eslint-config.